File: android/app/src/main/kotlin/com/typography/animations/entrance/SpiralDrop.kt
package com.typography.animations.entrance

import com.typography.AnimatedTextState
import com.typography.EasingType
import com.typography.animations.EasingFunctions

fun evaluateSpiralDrop(t: Float): AnimatedTextState {
    val elasticT = EasingFunctions.applyEasing(t, EasingType.ELASTIC_OUT)
    
    val scale = 0.2f + 0.8f * elasticT
    val rotation = -8.0f * (1.0f - elasticT)
    
    return AnimatedTextState(
        opacity = (t * 4.0f).coerceIn(0f, 1f),
        scale = scale,
        rotation = rotation
    )
}

File: lib/utils/animations/entrance/spiral_drop.dart
import '../base/animated_text_state.dart';
import '../easing/easing_functions.dart';
import '../../../models/editor_models.dart';

AnimatedTextState evaluateSpiralDrop(double t) {
  final elasticT = EasingFunctions.applyEasing(t, EasingType.elasticOut);
  
  final scale = 0.2 + 0.8 * elasticT;
  final rotation = -8.0 * (1.0 - elasticT); // Elegant micro-twist
  
  return AnimatedTextState(
    opacity: (t * 4.0).clamp(0.0, 1.0),
    scale: scale,
    rotation: rotation,
  );
}
